### PR TITLE
[MO] - [system test] -> Override bootstrap dns triggers rolling update

### PR DESCRIPTION
**Pull request by see-quick**
Signed-off-by: Seequick1 <morsak@redhat.com>

### Type of change

- Enhancement / new feature

### Description

This PR adding system test, which checks that `overridig bootstrap dns triggers rolling update.

### Checklist

- [x] Write tests
- [ ] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/RollingUpdateST.java [hunk 1]
```diff
@@ -131,6 +135,36 @@ void assertThatRollingUpdatedFinished(String rolledComponent, String stableCompo
...
+                            .withAddress(bootstrapDns)
+                        .endBootstrap()
+                    .endOverrides()
+                    .build());
+        });
+
+        StUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaPods);
+        StUtils.waitUntilKafkaCRIsReady(CLUSTER_NAME);
+
+        String bootstrapAddressDns = ((KafkaListenerExternalLoadBalancer) Crds.kafkaOperation(kubeClient().getClient())
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/RollingUpdateST.java]: This really just verifies that you updated the CR. Would it be worth to also check that the CO does what it is supposed to and the new bootstrap name is in the alternate DNS subjects in the certificates?

**Comment by scholzj** [comment]
@see-quick Is this still alive? Or should it be closed?

**Comment by see-quick** [comment]
So the main problem here is that I am missing the last step of the test, which is the verification that all messages are sent via the newly generated certificate or am I wrong?

**Comment by see-quick** [comment]
@scholzj

**Comment by scholzj** [comment]
That would assume the DNS name you configured actually exists and can be used. If that is the case, it could be the easiest solution - the TLS hsotname verification would veirfy for you that it is in the certificate. Other option is to just get the cert from the secret, decode it and check the new name is in the SANs.

File: systemtest/src/test/java/io/strimzi/systemtest/RollingUpdateST.java (hunk 1 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/RollingUpdateST.java]: Did it already )

**Comment by scholzj** [comment]
@ppatierno Do you think you can find some time to review this?

**Pull request merged**